Feat/create form

heroleggo opened the pull request:
## 작업 내용

- 숏폼 생성 로직 추가
- 해당 컨트롤러에 jwt guard 적용
- form-user 간 relation이 적용되어 있지 않은 부분 수정

heroleggo committed:
feat: Add create form logic

- 폼 생성 로직 추가
- form router에 auth guard 적용

heroleggo committed:
Merge branch 'dev' into feat/create-form

heroleggo committed:
fix: Update form domain codes

- req.user 사용 부분 수정
- form 에 user가 연결되어 있지 않던 부분 수정
- 이미 존재하는 투어 정보에 대해 폼 생성 테스트 완료

heroleggo committed:
Merge branch 'dev' into feat/create-form

heroleggo committed:
etc: Update verbose in file storage

- FileStorage 클래스에서 response를 무조건 출력해주던 부분 수정

heroleggo committed:
fix: Fix form service

- 폼 정보 조회 시 폼 생성 유저 정보도 조회되도록 수정

tndls2 commented on src/modules/forms/FormController.ts:
```diff
@@ -15,11 +31,31 @@ export class FormController {
     }
 
+    @Post('/')
+    @UseGuards(AuthGuard('jwt'))
```
모든 메소드에서 @UseGuards(AuthGuard('jwt')) 사용하는 거면 아예 컨트롤러에 가드 걸어도 괜찮을 거 같아욤

heroleggo commented:
### 추가 작업 예정 사항 (오늘 중)

- Controller 단위로 UseGuards 데코레이터 사용하도록 변경
- ffmpeg를 활용한 비디오 화질 변환 및 썸네일 추출 구현 시도

heroleggo committed:
Merge branch 'dev' into feat/create-form

heroleggo committed:
feat: Add thumbnail extractor on CommonService

- ffmpeg를 활용한 thumbnail 이미지 생성 및 S3 업로드 로직 추가
- 이 코드를 실행하기 위해서 ffmpeg 머신에 설치되어 있어야 함

heroleggo commented:
### 애 먹고 있는 사항 (진행되지 못함)

- 업로드된 숏폼 컨텐츠의 크기가 40MB 이상일 경우, 변환을 통해 사이즈를 낮추는 방법을 찾고 있었으나 비디오 별 화면 크기, 비율 등을 맞출 수가 없고, 변환했을 경우에 어떻게 화면이 바뀌는 지 알기가 어려워 우선 보류함

heroleggo committed:
Merge branch 'dev' into feat/create-form

heroleggo committed:
Merge branch 'dev' into feat/create-form

heroleggo committed:
Merge branch 'dev' into feat/create-form

heroleggo committed:
fix: Fix dto

- add decorator for validation
